ERR_MODULE_NOT_FOUND

### Describe the bug

There are sometime some errors while running the tests that happened only on the CI (GitHub).

The stack trace:

```shell
Error:  ⎯⎯⎯⎯⎯⎯ Failed Suites 1 ⎯⎯⎯⎯⎯⎯⎯
Error: 
Error:   FAIL  x.test.tsx [ x.test.tsx ]
Error:  Error: Cannot find module '/home/runner/actions-runner/_work/x/node_modules/.vitest/deps/react_jsx-runtime.js' imported from /home/runner/actions-runner/_work/x/node_modules/vite-node/dist/client.mjs
Error:  ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Error:  Serialized Error: { code: 'ERR_MODULE_NOT_FOUND' }
Error:  ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯
```

### Reproduction

Not possible.

### System Info

```shell
System:
    OS: macOS 14.2.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 73.50 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.17.1 - ~/.asdf/installs/nodejs/18.17.1/bin/node
    npm: 9.6.7 - ~/.asdf/plugins/nodejs/shims/npm
  Browsers:
    Brave Browser: 120.1.61.120
    Chrome: 120.0.6099.234
    Safari: 17.2.1
  npmPackages:
    @vitejs/plugin-react: 4.2.1 => 4.2.1 
    @vitest/coverage-v8: 1.2.1 => 1.2.1 
    @vitest/ui: 1.2.1 => 1.2.1 
    vite: 5.0.12 => 5.0.12 
    vitest: 1.2.1 => 1.2.1
```


### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/vitest/src/node/plugins/utils.ts
+++ b/packages/vitest/src/node/plugins/utils.ts
@@ -34,7 +34,7 @@ export function resolveOptimizerConfig(_testOptions: DepsOptimizationOptions | u
       'vue',
       ...(testOptions.exclude || viteOptions?.exclude || []),
     ]
-    const runtime = currentInclude.filter(n => n.endsWith('jsx-dev-runtime'))
+    const runtime = currentInclude.filter(n => n.endsWith('jsx-dev-runtime') || n.endsWith('jsx-runtime'))
     exclude.push(...runtime)
 
     const include = (testOptions.include || viteOptions?.include || []).filter((n: string) => !exclude.includes(n))

